### PR TITLE
Bad credentials

### LINKED ISSUE #64: User Story 12, Users who are logged in already are redirected
```
As a registered user, merchant, or admin
When I visit the login path
If I am a regular user, I am redirected to my profile page
If I am a merchant user, I am redirected to my merchant dashboard page
If I am an admin user, I am redirected to the home page of the site
And I see a flash message that tells me I am already logged in
```

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written (if required)
- [x] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #67

child of #79

### LINKED ISSUE #65: User Story 11, User cannot log in with bad credentials
```
As a visitor
When I visit the login page ("/login")
And I submit invalid information
Then I am redirected to the login page
And I see a flash message that tells me that my credentials were incorrect
I am NOT told whether it was my email or password that was incorrect
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #67

child of #79

depends on #66

### PR DESCRIPTION
**Pull request by teresa-m-knowles**
As a registered user, merchant, or admin
When I visit the login path
If I am a regular user, I am redirected to my profile page
If I am a merchant user, I am redirected to my merchant dashboard page
If I am an admin user, I am redirected to the home page of the site
And I see a flash message that tells me I am already logged in

Closes #65 
Closes #64 

Co authored with @timnallen

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 class SessionsController < ApplicationController
+
   def new
+    if current_user
```
Comment (allieIsOkay) [inline app/controllers/sessions_controller.rb:4]: We had to write something very similar for the login routing at first, it might be useful to make a helper method to handle these redirects in order to remove the duplication.

**Review by allieIsOkay — APPROVED** [review]

**Comment by csvlewis** [comment]
We just went over this together in person, looks good!
